core/src/main/java/greencity/validator/EventDtoRequestValidator.java

```diff
@@ -6,49 +6,89 @@
+                || eventDateLocationDtos.size() > ValidationConstants.MAX_EVENT_DATES_AMOUNT) {
+                throw new EventDtoValidationException(ErrorMessage.WRONG_COUNT_OF_EVENT_DATES);
+            }
 
-        for (var eventDateLocationDto : eventDateLocationDtos) {
-            if (eventDateLocationDto.getStartDate().isBefore(ZonedDateTime.now())
-                || eventDateLocationDto.getStartDate().isAfter(eventDateLocationDto.getFinishDate())) {
-                throw new EventDtoValidationException(ErrorMessage.EVENT_START_DATE_AFTER_FINISH_DATE_OR_IN_PAST);
+            for (var eventDateLocationDto : eventDateLocationDtos) {
+                if (!validateEventDateLocation(eventDateLocationDto, context)) {
+                    return false;
+                }
+
+                if (eventDateLocationDto.getStartDate().isBefore(ZonedDateTime.now())
+                    || eventDateLocationDto.getStartDate().isAfter(eventDateLocationDto.getFinishDate())) {
+                    throw new EventDtoValidationException(ErrorMessage.EVENT_START_DATE_AFTER_FINISH_DATE_OR_IN_PAST);
+                }
+
+                AddressDto addressDto = eventDateLocationDto.getCoordinates();
+                String onlineLink = eventDateLocationDto.getOnlineLink();
+                if (onlineLink == null && addressDto == null) {
+                    throw new EventDtoValidationException(ErrorMessage.NO_EVENT_LINK_OR_ADDRESS);
+                }
+                if (onlineLink != null) {
+                    isUrlValid(onlineLink);
+                }
             }
-            AddressDto addressDto = eventDateLocationDto.getCoordinates();
-            String onlineLink = eventDateLocationDto.getOnlineLink();
-            if (onlineLink == null && addressDto == null) {
-                throw new EventDtoValidationException(ErrorMessage.NO_EVENT_LINK_OR_ADDRESS);
+
+            if (addEventDtoRequest.getTags().size() > ValidationConstants.MAX_AMOUNT_OF_TAGS) {
+                throw new EventDtoValidationException(ErrorMessage.WRONG_COUNT_OF_TAGS_EXCEPTION);
             }
-            if (onlineLink != null) {
-                isUrlValid(onlineLink);
+
+            return true;
+        } else if (value instanceof UpdateEventDto) {
+            UpdateEventDto updateEventDto = (UpdateEventDto) value;
+            List<EventDateLocationDto> eventDateLocationDtos = updateEventDto.getDatesLocations();
+
+            for (var eventDateLocationDto : eventDateLocationDtos) {
+                if (!validateEventDateLocation(eventDateLocationDto, context)) {
+                    return false;
+                }
             }
+
+            return true;
         }
 
-        if (value.getTags().size() > ValidationConstants.MAX_AMOUNT_OF_TAGS) {
-            throw new EventDtoValidationException(ErrorMessage.WRONG_COUNT_OF_TAGS_EXCEPTION);
+        return false;
+    }
+
+    private boolean validateEventDateLocation(EventDateLocationDto eventDateLocationDto,
```

Comment:
It would be more understandable if you began the boolean-returning method’s name with “isValid” instead of “validate.”

Reply:
changed the method name